compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12810,6 +12810,149 @@ TR::Node *su2dSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    return node;
    }
 
+
+/**
+ * \brief
+ * Remove add/subtract operations under comparison nodes to save an
+ * addition or subtraction instruction during codegen. This simplification
+ * applies to comparisons of byte, short, int, and long values.
+ *
+ * \details
+ *
+ * \verbatim
+ * As an example, assuming that the following is the original expression
+ *        x + oldConst1 > oldConst2
+ *
+ * This expression can be simplified to
+ *        x > newConst
+ *        where newConst = oldConst2 - oldConst1
+ *
+ * \endverbatim
+ *
+ *  And the newConst calculation should not produce overflow or underflow.
+ *
+ * \param node          the comparison node being simpilfied
+ * \param s             the simplifier object
+*/
+TR::Node* removeArithmeticsUnderIntegralCompare(TR::Node* node,
+                                                TR::Simplifier * s)
+   {
```

Comment:
`removeArithmeticsUnderIntegralCompare()` is only invoked in byte, short, integer or long comparison simplifer handlers. Besides, it tests the first child of the `node` to see if it's an `iadd`, `isub`, and etc.  So I don't think we need to test the comparison node.

Reply:
I think an assert is probably warranted as an additional safety measure just in case the function use is changed or abused later. Use an assert that is not enabled in prod builds would be my vote.

Reply:
I agree. If you make an implicit assumption, enforce it with a `TR_ASSERT`.